mono/metadata/object-internals.h

```diff
@@ -2143,12 +2143,15 @@ mono_runtime_object_init_checked (MonoObject *this_obj, MonoError *error);
 MONO_PROFILER_API MonoObject*
 mono_runtime_try_invoke (MonoMethod *method, void *obj, void **params, MonoObject **exc, MonoError *error);
 
-// The exc parameter is deliberately missing and so far this has proven to reduce code duplication.
+// The exc parameter was deliberately missing and so far this has proven to reduce code duplication.
 // In particular, if an exception is returned from underlying otherwise succeeded call,
 // is set into the MonoError with mono_error_set_exception_instance.
 // The result is that caller need only check MonoError.
+//
+// FIXME The removal of exc parameter confuses ongoing coop conversion. It has been restored.
+// Prior conversion should be revisited (despite long term success).
 MonoObjectHandle
-mono_runtime_try_invoke_handle (MonoMethod *method, MonoObjectHandle obj, void **params, MonoError* error);
+mono_runtime_try_invoke_handle (MonoMethod *method, MonoObjectHandle obj, void **params, MonoObjectHandleOut exc, MonoError* error);
```

Comment:
I liked the fixes of the memory leaks in `mono_attach_load_agent`.

1. I don't understand why the seemingly unrelated change to `mono_runtime_try_invoke_handle` is also in this PR.  What am I missing?

2. We _want_ all exceptions to go via `MonoError` in this function.  Having both a `MonoError` and an exception out-arg is a step back.

Reply:
`mono_runtime_try_invoke_handle`

This is a confusing thing agreed.
I've been round and round on this, at least with myself.

In the past I have come upon `mono_runtime_try_invoke`
and a need to convert pointer to handle, at least twice.

The first time I looked closely, removed the exc parameter,
and convinced myself it was semantica preserving.

The second time I was again puzzled, looked closely, and
remained convinced it was a correct change.

Here we are a third time.
Again the code is not obvious and I get tired of trying
to prove to myself what it does.

I suggest a conversion should have and now can be done
that more closely matches the original.

I believe the old conversion should be revisited.

This PR converts the new case, but does not yet reconvert the old ones.

You say it is a step back. Maybe you are right.
But I find it is a confusing change and would prefer to preserve
the old semantics, assuming they were ok.